bin/aurbuild

```diff
@@ -65,32 +70,50 @@ source /usr/share/makepkg/util.sh
 
 while getopts :a:cd:p:r: OPT; do
     case $OPT in
-        a|+a) queue="$OPTARG"    ;;
-        c|+c) build=build_chroot ;;
-        d|+d) database="$OPTARG" ;;
-        p|+p) pool="$OPTARG"     ;;
-        r|+r) root="$OPTARG"     ;;
-        *) plain "usage: $argv0 [-c] -a <queue> -d <database> -p <pool> -r <root> [--] [makepkg args]"
-           exit 1 ;;
+        a) queue="$OPTARG"    ;;
+        c) build=build_chroot ;;
+        d) database="$OPTARG" ;;
+        p) pool="$OPTARG"     ;;
+        r) root="$OPTARG"     ;;
+        :) case $OPTARG in
+                a) error "$argv0: No package queue specified (-a)"; usage; exit 1 ;;
+                d) error "$argv0: No repository database specified (-d)"; usage; exit 1 ;;
+                p) error "$argv0: No package pool specified (-p)"; usage; exit 1 ;;
+                r) error "$argv0: No repository root specified (-r)"; usage; exit 1 ;;
+            esac ;;
+        ?) error "$argv0: Option not recognized (-$OPTARG)"; usage; exit 1; ;;
     esac
 done
 
+# Validate arguments
+re_blank='^[[:space:]]*$'
```

Comment:
It seems odd to report "Missing arguments" here, when it should already be done in the getopts loop above. I assume this is working around a getopts limitation?